nox/registry.py

```diff
@@ -2,23 +2,49 @@
 
 import collections
 import copy
+import functools
 
+import attr
 
 _REGISTRY = collections.OrderedDict()
 
 
-def session_decorator(func):
+@attr.s
+class PythonConfig:
+    python = attr.ib(default=None)
+    virtualenv = attr.ib(default=None)
+    reuse = attr.ib(default=None)
+
+
+def _to_python_config(value):
+    if not isinstance(value, PythonConfig):
+        return PythonConfig(python=value)
+    return value
+
+
+def session_decorator(func=None, python=None):
     """Designate the decorated function as a session."""
+    if func is None:
```

Comment:
This is an unusual enough idiom that is worth explaining:

```python
# If `func` is provided, then this is the decorator call with the function
# being sent as part of the Python syntax (`@nox.session`).
# If `func` is None, however, then this is a plain function call, and it
# must return the decorator that ultimately is applied (`@nox.session(...)`).
#
# This is what makes the syntax with and without parentheses both work.
```

Reply:
Good call. Done.